operational_analysis/toolkits/reanalysis_downloading.py

```diff
@@ -0,0 +1,369 @@
+            density are computed from the downloaded reanalysis variables. Note that this requires the downloaded
+            eastward wind speed, northward wind speed, temperature, and surface pressure to be renamed to the standard
+            variable names "u_ms", "v_ms", "temperature_K", and "surf_pres_Pa". Defaults to False.
+        save_pathname (:obj:`string`, optional): The path where the downloaded reanalysis data will be saved (if
+            defined). Defaults to None.
+        save_filename (:obj:`string`, optional): The file name used to save the downloaded reanalysis data (if
+            defined). Defaults to None.
+        apikey_file (:obj:`string`, optional): The combined path and file name where the PlanetOS API key is saved. If
+            undefined, will assume there is a file called "APIKEY" saved in the operational_analysis.toolkits
+            directory. Defaults to None.
+
+    Returns:
+        :obj:`pandas.DataFrame`: A dataframe containing time series of the requested reanalysis variables
+    """
+
+    # Import PlanetOS apikey
+    if apikey_file is None:
+        apikey_file = Path(__file__).parent.resolve() / "APIKEY"
+    apikey = open(apikey_file).read().strip()
+
+    # Get start and end dates of dataset
+    start_date_ds, end_date_ds = get_dataset_start_end_dates_planetos(dataset, apikey)
+
+    # Depending on the start date, end date, and number of years specified, calculate the appropriate start and end
+    # dates
+    start_date, end_date = _get_start_end_dates_planetos(
+        start_date, end_date, num_years, start_date_ds, end_date_ds
+    )
+
+    # Determine number of data points
+    date_diff = end_date - start_date
+    count = int(24 * date_diff.days + date_diff.seconds / 3600)
+
+    # Get default variable names and dictionary if not defined
+    if var_names is None:
+        var_dict = _get_default_var_dicts_planetos(dataset)
+        var_names = list(var_dict.keys())
+
+    # Download data from PlanetOS
+    url = "http://api.planetos.com/v1/datasets/" + dataset_names[dataset] + "/point?"
+
+    kwgs = {
+        "apikey": apikey,
+        "count": count,
+        "lon": lon,
+        "lat": lat,
+        "vars": ", ".join(var_names),
+        "time_start": start_date.strftime("%Y-%m-%dT%H:%M:%S"),
+        "time_end": end_date.strftime("%Y-%m-%dT%H:%M:%S"),
+    }
+
+    r = requests.get(url, params=kwgs)
+
+    # convert to standard dataframe
+    df = _convert_resp_to_df(r, var_names, var_dict)
+
+    # compute derived variables if requested
+    if calc_derived_vars:
+        df["windspeed_ms"] = np.sqrt(df["u_ms"] ** 2 + df["v_ms"] ** 2)
+        df["winddirection_deg"] = met.compute_wind_direction(df["u_ms"], df["v_ms"])
+        df["rho_kgm-3"] = met.compute_air_density(df["temperature_K"], df["surf_pres_Pa"])
+
+    if save_filename is not None:
+        df.to_csv(os.path.join(save_pathname, save_filename + ".csv"), index=False)
```

Comment:
This comment is only because I noticed you used the `Path` object previously, and this is now the only use of the `os` library, so this could be amended to the following and we can remove an import from the top. That said, I see this is optional, and up to you. 
```python
df.to_csv(Path(save_pathname).resolve() / f"{save_filename}.csv", index=False)
```

Reply:
Good call, I made this change and removed `os` from the imports